libs/stats/odc/stats/_cli_common.py

```diff
@@ -92,12 +92,25 @@ def click_yaml_cfg(*args, **kw):
     """
     def _parse(ctx, param, value):
         if value is not None:
-            from odc.io.text import parse_yaml_file_or_inline
-
-            try:
-                return parse_yaml_file_or_inline(value)
-            except Exception as e:
-                raise click.ClickException(str(e)) from None
+            import urllib
+            from urllib.parse import urlparse
+            r = urlparse(value)
+
+            # if referece to file, use documents.load_documents to process
+            if all([r.scheme, r.netloc]):
+                from datacube.utils import documents
+                try:
+                    return dict(next(documents.load_documents(value)))
+                except urllib.error.URLError as e:
+                    raise click.ClickException(str(e) + f". Cannot access file {value}") from None 
+                except Exception as e:
+                    raise click.ClickException(str(e)) from None 
+            else:
+                from odc.io.text import parse_yaml_file_or_inline
+                try:
+                    return parse_yaml_file_or_inline(value)
```

Comment:
I think this whole section can be removed. Also, there's no need to use `parse_yaml_file_or_inline()` I don't think, since I don't think it makes any sense to try to read a literal yaml file from a CLI argument.

Reply:
I am OK to remove the support `read a literal yaml file from a CLI argument`. 

But it's a shared method, and should let stakeholders know it: @alexgleith @Kirill888

Reply:
I found it useful for embedding config in k8s, but also I don't care.

Reply:
We embed config in k8s so please don't change the current behaviour.